### DISCUSSION TITLE
Stiffness tensor of void or bubble when coupling elastic energy

### DISCUSSION THREAD

**Discussion by wenpeng231024** [idaholab/moose#18288] [Q&A General] [answered]
Dear MOOSE group,

I try to add elastic energy in our model of void or bubble evolution in the matrix as Aagesen's paper (Aagesen, 2017, Quantifying elastic energy effects on interfacial energy in the Kim-Kim-Suzuki phase-field model with different interpolation schemes). 

https://www.sciencedirect.com/science/article/pii/S0927025617304111

However, I am a little confused about how to set the void/bubble stiffness tensor. I found most papers assuming the stiffness tensor of gas bubbles is the same as the matrix. Does this make sense? The stress due to the eigenstrain is constant. But the elastic strain should be different on the interface using different stiffness tensors of gas bubbles. And the stress in the matrix caused by the void/bubble should be also different. 

So my question is that how to select the stiffness tensor of void/bubble when we use eigenstrain to develop stress in the matrix caused by the void/bubble?

I also show some relevant papers here.

Li, Nuclear Instruments and Methods in Physics Research B 303 (2013) 62–67. 
Hu, Journal of Nuclear Materials 392 (2009) 292–300.
Xiao, Computational Materials Science 184 (2020) 109867.

Thanks,

Peng

**Comment by GiudGiud** [comment, marked as answer]
@laagesen

**Reply by laagesen** [reply]
Hi Peng, to use the eigenstrain approach to simulate gas bubble pressure, I think it's fine to use the bubble stiffness tensor to be the same as the matrix. What you are doing is using an inclusion to mimic the effect of the gas pressure, which can result in a non-zero stress experienced by the matrix outside of the bubble. (The pressure felt outside the bubble is actually given by the difference between the gas bubble pressure and the surface tension. This is not considered in either of the two earlier works you mentioned, but it is considered in the 2020 paper.) So using the eigenstrain approach will allow you to consider a bubble with a fixed overpressure or underpressure relative to surface tension. (For an equilbrium bubble, or a bubble whose pressure is exactly balanced by surface tension, there is no stress felt outside the bubble.) So the bubble phase elastic constant do not physically represent the actual elastic constants of the gas phase; this is sort of a trick to mimic the effect of the gas pressure on the surrounding matrix. This approach has some disadvantages, such as inaccurate response to applied loads.

**Reply by wenpeng231024** [reply]
Hi Larry,

Thanks so much for your reply. If I understand correctly, that means the calculated total stress or elastic stress inside the bubble is meaningless. Because we just care about the stress in the matrix around the bubble. 

Actually, I try to test using different elastic constants of gas bubble. I found when I use VTS interpolation scheme (assuming the total strain is the same for both phases on the interface) to add the elastic energy, a larger elastic stress is obtained in the matrix around the bubble using a smaller elastic constants of gas bubble. Because, although the pressure in the bubble is a constant. But a larger total strain (Total strain = elastic strain - eigenstrain) for bubble is obtained when using a smaller stiffness of gas bubble. (For gas, it seems more reasonable to use a smaller elastic modulus.) This larger total strain in the bubble leads to a larger strain in the matrix through the interface due to the continuum assumption.  

So how can we remove this stress difference in the matrix due to the selected of the stiffness of gas bubble? Maybe it's better to use KHS scheme? Because it assume the stress is the same for both phases on the interface. Or we can ignore this effect if this difference is not very large.

You mentioned this approach is inaccurate response to applied stress in the matrix. This is exactly what I am doing. Can you show more about this? I use "ADPressure" boundary conditions to add applied stress in the matrix. Then, I compared the stress in the matrix around the bubble between adding applied stress case and no applied stress case. Their difference is approximately the applied stress. Does this indicate the eigenstrain approach works for applied loading?

Many thanks,

Peng

**Reply by laagesen** [reply]
Hm that is interesting. I do think that seems reasonable for the applied stress case, for what you are looking at (the stress in the matrix). I was thinking more about applying a load and looking at the total displacement in response to that load, for example. If you apply a fixed load, for the case of a void, physically there is nothing to resist the applied load, so you would expect a larger total displacement than for the simulations where you mimic the effect of a void with the eigenstrain approach, because the void is represented by a region of material that has the same elastic constant as the matrix and therefore significantly resists the applied load.

**Reply by wenpeng231024** [reply]
Yes, I agree with you about the void. 

I think the elastic constants of void/bubble should affect the stress in the matrix. And the applied loading maybe enlarge this effect. 

Thanks,

Peng
